File: algorithms/length_of_last_word.c
#include <stdio.h>

// https://leetcode.com/problems/length-of-last-word/

int length_of_last_word(char *s);


int main()
{
    printf("%d\n", length_of_last_word(""));
    printf("%d\n", length_of_last_word("Hello World"));
    return 0;
}

int length_of_last_word(char *s)
{
    int length = 0;

    while(*s) s++;
    s--;
    while(*s && ' ' == *s) s--;

    while(*s && ' ' != *s){
        s--;
        length++;
    }

    return length;
}

File: algorithms/length_of_last_word.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


# https://leetcode.com/problems/length-of-last-word/


def length_of_last_word(s):
    s = s.strip()
    return len(s.split()[-1]) if s else 0


if __name__ == '__main__':
    print length_of_last_word("Hello World")
    print length_of_last_word("  ")
